compiler/rustc_mir_dataflow/src/move_paths/builder.rs

```diff
@@ -227,10 +227,13 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
                     }
                     _ => bug!("Unexpected type {place_ty:#?}"),
                 },
-                // `OpaqueCast` only transmutes the type, so no moves there and
-                // `Downcast` only changes information about a `Place` without moving
+                // `OpaqueCast`:Only transmutes the type, so no moves there.
+                // `Downcast`  :Only changes information about a `Place` without moving.
+                // `Subtype`   :Even though we have move here, it doesn't effect behavior of the code.
```

Comment:
```suggestion
                // `Subtype`   :Only transmutes the type, so no moves there.
```
shouldn't this be the same as `OpaqueCast`?